src/lib/modals/loadModalsOnRouteChange.ts

```diff
@@ -0,0 +1,23 @@
+import { useDb } from '../db'
+import { useRouter } from '../router/store'
+import { useEffect, useRef } from 'react'
+import { useModals } from './store'
+import { usePreferences } from '../preferences'
+
+export default function useLoadModalsOnRouteChangeHandler() {
+  const { pathname } = useRouter()
+  const db = useDb()
+  const { openModals } = useModals()
+  const { preferences } = usePreferences()
+
+  const storageMapRef = useRef(db.storageMap)
+  useEffect(() => {
```

Comment:
This must be called only once. opoenModals still can be called. https://github.com/streamich/react-use/blob/master/docs/useMount.md should be better.